db/migrate/20210616124359_create_subject_specialisms.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateSubjectSpecialisms < ActiveRecord::Migration[6.1]
+  def change
+    create_table :subject_specialisms do |t|
+      t.string :name, null: false, index: { unique: true }
+      t.references :allocation_subject, index: true, null: false, foreign_key: { to_table: :allocation_subjects }
```

Comment:
```suggestion
      t.references :allocation_subject, null: false, foreign_key: true
```

Reply:
`to_table` is only needed if the name of the foreign key was different from the rails convention.

Reply:
fixed